lsp/nls/src/linearization/mod.rs

```diff
@@ -437,7 +437,14 @@ impl Linearizer for AnalysisHost {
             scope,
         } = lin.into_inner();
 
-        linearization.sort_by_key(|item| (item.pos.src_id, item.pos.start));
+        linearization.sort_by(|it1, it2| match (it1.pos, it2.pos) {
+            (TermPos::None, TermPos::None) => std::cmp::Ordering::Equal,
+            (TermPos::None, _) => std::cmp::Ordering::Less,
+            (_, TermPos::None) => std::cmp::Ordering::Greater,
+            (pos1, pos2) => (pos1.unwrap().src_id, pos1.unwrap().start)
+                .partial_cmp(&(pos2.unwrap().src_id, pos2.unwrap().start))
```

Comment:
Why do we need a `partial_cmp` that is then unwrapped? Doesn't a tuple `(T, U)` implements `Ord` if `T: Ord, U: Ord` ?